### PR TITLE
Add a docstring for the AutoencoderKL's encode

### LINKED ISSUE #5229: [HacktoberFest][Missing Docstrings] Autoencoder `encode`
This feature request is included in this years Diffusers' Hacktoberfest 👨‍💻! 

Hacktoberfest's purpose is to give upcoming open-source developers the opportunity to learn about open-source and programming by fixes simple issues (like this one).

**How to solve the issue**:

- 1. Every issue has a description further below explaining the problem. First, you should try to understand the issue.
- 2. Second, follow the [contribution guide](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md) to learn how to contribute to Diffusers.\
- 3. Claim this issue for you by opening a PR that links to this issue and writing "I am solving this issue in `<link-to-pr>`"

**Issue**

The [encode method of AutoencoderKL](https://github.com/huggingface/diffusers/blob/78a78515d64736469742e5081337dbcf60482750/src/diffusers/models/autoencoder_kl.py#L252) is completely missing docstrings, can you add them?

### PR DESCRIPTION
**Pull request by freespirit**
Fixes #5229

### CODE REVIEW

**Comment by freespirit** [comment]
I tried to reuse as much as possible the style and contents of the `tiled_encode` docstring

**Review by patrickvonplaten — APPROVED** [review]
Looks great to me :heart:

File: src/diffusers/models/autoencoder_kl.py [hunk 1]
```diff
@@ -249,7 +249,21 @@ def set_default_attn_processor(self):
         self.set_attn_processor(processor, _remove_lora=True)
 
     @apply_forward_hook
-    def encode(self, x: torch.FloatTensor, return_dict: bool = True) -> AutoencoderKLOutput:
+    def encode(
+        self, x: torch.FloatTensor, return_dict: bool = True
+    ) -> Union[AutoencoderKLOutput, tuple[DiagonalGaussianDistribution]]:
```
Comment (patrickvonplaten) [inline src/diffusers/models/autoencoder_kl.py:254]:
```suggestion
    ) -> Union[AutoencoderKLOutput, Tuple[DiagonalGaussianDistribution]]:
```

to support Python 3.8 syntax

**Comment by freespirit** [comment]
Ran `make style` & `make quality`. Thank you all for the support!

**Comment by patrickvonplaten** [comment]
Failing tests are unrelated - merging!

**Pull request merged**